packages/sdk-js/src/GrowthBook.ts

```diff
@@ -726,20 +743,33 @@ export class GrowthBook<
               id: experiment.key,
             }
           );
+          this._unsetAntiFlicker();
           return result;
         }
         this._redirectedUrl = url;
         const navigate = this._getNavigateFunction();
         if (navigate) {
           if (isBrowser) {
-            this._setAntiFlicker();
-            window.setTimeout(() => {
+            this._setAntiFlicker(true);
+            // Wait for the possibly-async tracking callback, bound by min and max delays
+            Promise.all([
+              promiseTimeout(
+                this._activeTrackingCall,
+                this._ctx.maxNavigateDelay ?? 3000
+              ),
+              new Promise((resolve) =>
+                window.setTimeout(resolve, this._ctx.navigateDelay ?? 300)
+              ),
+            ]).then(() => {
               try {
-                navigate(url);
+                if (!this._isNavigating) {
+                  this._isNavigating = true;
```

Comment:
This needs to get unset after the navigation finishes.  Especially important for SPAs where the navigate callback doesn't do a full page load.

Reply:
Do you mean something like
```
navigate(url).then(() => this._isNavigating = false);
```
One possible problem is that navigate() may not be async or at least not trackable (such as `window.location.replace(url)`), so it would get unset instantly. Maybe that's ok though, but then it doesn't do much to dedupe.

Or perhaps we make the default navigate function return a promise with an arbitrary 1sec timeout?

Reply:
move anti flicker to _updateAllAutoExperiments, outside of loop

Reply:
no anti-flicker in constructor, just as part of the (default) navigate function.

Reply:
no anti-flicker for dom changes. only for default navigate function. make sure hooks work...